marklogic-data-hub-central/ui/src/pages/Browse.tsx

```diff
@@ -160,23 +160,24 @@ const Browse: React.FC<Props> = ({ location }) => {
       }
       applySaveQuery(options);
     }
-    if(location.state && location.hasOwnProperty('zeroState') && !location.state['zeroState']){
-        setPageWithEntity(location.state['entity'],
-            location.state['pageNumber'],
-            location.state['start'],
-            location.state['searchFacets'],
-            location.state['query'])
-        location.state['tableView'] ? toggleTableView(true) : toggleTableView(false);
-    }
-    if(location.state && location.state['entityName'] && location.state['jobId']){
-        setLatestJobFacet(location.state['jobId'], location.state['entityName']);
+
+    if (location.state && location.state['entityName'] && location.state['jobId']) {
+      setLatestJobFacet(location.state['jobId'], location.state['entityName']);
     }
     if (location.state && location.state['entity']) {
       setEntityClearQuery(location.state['entity']);
     }
     if (location.state && location.state['jobId']) {
       setLatestJobFacet(location.state['jobId'], location.state['entityName']);
     }
+    if (location.state && !location.state['zeroState']) {
```

Comment:
@timur-isangulov Removing hasOwnProperty condition will throw an error when user clicks on last processed time from modelling screen

Reply:
A very solid reason to add RTL tests for Browse page.

Reply:
Hi Sanjeevani, thank you for testing and noticing this case. I changed the logic can you please review again.